include/swift/Reflection/TypeRefBuilder.h

```diff
@@ -760,6 +839,40 @@ class TypeRefBuilder {
   /// Get the unsubstituted capture types for a closure context.
   ClosureContextInfo getClosureContextInfo(RemoteRef<CaptureDescriptor> CD);
 
+  /// Get the multipayload enum projection information for a given TR
+  RemoteRef<MultiPayloadEnumDescriptor> getMultiPayloadEnumInfo(const TypeRef *TR);
+
+private:
+  llvm::Optional<uint64_t> multiPayloadEnumPointerMask;
+
+public:
+  /// Retrieve the MPE pointer mask from the target
+  // If it can't read it, it will make an educated guess
+  // Note: This is a pointer-sized value stored in a uint64_t
+  // If the target is 32 bits, the mask is in the lower 32 bits
+  uint64_t getMultiPayloadEnumPointerMask() {
```

Comment:
Here's the code in TypeRefBuilder to come up with the Enum pointer mask.  This first queries the target via `getSymbolAddress`.  But because tools like swift-reflection-dump don't implement `getSymbolAddress`, I need a fallback with logic similar to what I had before.

Reply:
Looks good!